pkg/reconciler/metrics/metrics.go

```diff
@@ -0,0 +1,212 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/tektoncd/triggers/pkg/client/listers/triggers/v1alpha1"
+	"github.com/tektoncd/triggers/pkg/client/listers/triggers/v1beta1"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"k8s.io/apimachinery/pkg/labels"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/metrics"
+)
+
+var (
+	elMetricsName = "eventlistener_count"
+	elCount       = stats.Float64(elMetricsName,
+		"number of eventlistener",
+		stats.UnitDimensionless)
+	elCountView *view.View
+
+	tbMetricsName = "triggerbinding_count"
+	tbCount       = stats.Float64(tbMetricsName,
+		"number of triggerbinding",
+		stats.UnitDimensionless)
+	tbCountView *view.View
+
+	ctbMetricsName = "clustertriggerbinding_count"
+	ctbCount       = stats.Float64(ctbMetricsName,
+		"number of clustertriggerbinding",
+		stats.UnitDimensionless)
+	ctbCountView *view.View
+
+	ttMetricsName = "triggertemplate_count"
+	ttCount       = stats.Float64(ttMetricsName,
+		"number of triggertemplate",
+		stats.UnitDimensionless)
+	ttCountView *view.View
+
+	ciMetricsName = "clusterinterceptor_count"
+	ciCount       = stats.Float64(ciMetricsName,
+		"number of clusterinterceptor",
+		stats.UnitDimensionless)
+	ciCountView *view.View
+)
+
+type listers struct {
+	elListener  v1beta1.EventListenerLister
```

Comment:
minor: these should be `elLister` not `elListener`
Or even better we could just call them `listers.el.` etc.

Reply:
Yes, `listers.el` make more sense. Changing to it.